sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -20,10 +20,30 @@ def valid_key?(key)
         key.instance_of?(String)
       end
 
-      def valid_value?(value)
+      def valid_primitive_value?(value)
```

Comment:
I don't think it's accurate to refer to any values in ruby as being primitive since all are objects.

Reply:
I introduced the term "primitive" because it seemed like an easy way to add some readability by distinguishing between composite types (arrays) and everything else. Do you think there's a better term for use? I'm just drawing "primitive" from some quick Googling (cf https://en.wikipedia.org/wiki/List_of_data_structures#Data_types) but it's possible there's a better term.

Absent that, I suppose this could just be inlined into `valid_value?` but that seems like a step back in readability.

Reply:
I totally understand the intent and my main issue is that it means something else in most other programming languages. Otherwise, it's a really good word for the purpose. I would be fine with inlining as it saves us from having to come up with a better name (the hardest problem in CS). Some suggestions for alternative names, if we want to go that route, would be `...basic_value` or `...simple_value`. I'm not crazy about them, but they'd work.